wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+		 *
+		 * @see https://github.com/xwp/pwa-wp/issues/17#issuecomment-406188455
+		 */
+		/* Translators: %s: a link for more details */
+		$insecure_content_description = esc_html__( 'Your home page doesn’t contain insecure URLs. However, there may be URLs on other pages that could be blocked. %s', 'pwa' );
+
+		/*
+		 * Todo: change !== to === as this is only like this for development.
+		 * The WP_HTTPS_Detection doesn't work with my local SSL certificate.
+		 * Also, change $this->is_currently_https() to ! $this->is_currently_https().
+		 * This is also for development only.
+		 * This main if block should only run if the site can use HTTPS, but the 'home' and 'siteurl' options aren't HTTPS.
+		 */
+		if ( $this->is_currently_https() && true !== get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) ) :
+			?>
+			<p class="description">
+				<?php
+				echo wp_kses_post( sprintf(
+					/* Translators: %s: a link for more details */
+					esc_html__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to HTTPS on your site. %s', 'pwa' ),
+					$https_more_details
+				) );
+				?>
+			</p>
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'HTTPS Upgrade', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( $upgrade_https_value, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_https_value ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description"><?php esc_html_e( 'Your site appears to support HTTPS', 'pwa' ); ?></p>
+
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'Upgrade Insecure URLs', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( $upgrade_insecure_content, self::OPTION_SELECTED_VALUE ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_INSECURE_CONTENT_OPTION ); ?>" type="radio" <?php checked( self::OPTION_SELECTED_VALUE !== $upgrade_insecure_content ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
+			</p>
+			<p class="description">
+				<?php echo wp_kses_post( sprintf( $insecure_content_description, $insecure_content_more_details ) ); ?>
+			<p>
+			<?php
+		else :
+			/* Translators: %s: HTTPS more details link */
+			echo wp_kses_post( sprintf( __( 'Your site doesn\'t look to support HTTPS. %s', 'pwa' ), $https_more_details ) );
+		endif;
+	}
+
+	/**
+	 * Whether the options indicate that the site is currently using HTTPS.
+	 *
+	 * @return boolean
```

Comment:
Per https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/ the `@return` should:

> contain all possible return types, and a description for each. Use a period at the end.

Also `bool` should be used instead of `boolean`. So this should be:

```
@return bool Whether currently HTTPS.
```

Reply:
Thanks, [fbd258](https://github.com/xwp/pwa-wp/pull/36/commits/fbd25818243cbd6f763b7b7406c318c9ba64ba20) changes the `@return` tag.